src/bin/matrix-max.cc

```diff
@@ -0,0 +1,121 @@
+        "combination methods). If number of rows differ in matrices, crop to\n"
+        "smallest one. A spk2utt-like file must be provided to specify which\n"
+        "sets of matrices to combine.\n"
+        "Usage: matrix-max <spk2utt-rspecifier> <matrix-rspecifier> "
+        "<matrix-wspecifier>\n"
+        "e.g.: matrix-max ark:data/spk2utt scp:exp/output.scp ark:exp/outputs_max.ark\n";
+
+    ParseOptions po(usage);
+    bool binary_write = false;
+    po.Register("binary", &binary_write, "If true, write output in binary "
+                "(only applicable when writing files, not archives/tables).");
+
+    po.Read(argc, argv);
+
+    if (po.NumArgs() != 3) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    std::string spk2utt_rspecifier = po.GetArg(1),
+        matrix_rspecifier = po.GetArg(2),
+        matrix_wspecifier = po.GetArg(3);
+
+    int64 num_spk_done = 0, num_spk_err = 0,
+        num_utt_done = 0, num_utt_err = 0;
+
+    RandomAccessBaseFloatMatrixReader matrix_reader(matrix_rspecifier);
+    SequentialTokenVectorReader spk2utt_reader(spk2utt_rspecifier);
+    BaseFloatMatrixWriter matrix_writer(matrix_wspecifier);
+
+    for (; !spk2utt_reader.Done(); spk2utt_reader.Next()) {
+      std::string spk = spk2utt_reader.Key();
+      const std::vector<std::string> &uttlist = spk2utt_reader.Value();
+      if (uttlist.empty()) {
+        KALDI_ERR << "Speaker with no utterances.";
+      }
+      Matrix<BaseFloat> spk_max;
+      int32 utt_count = 0;
+
+      // All matrices may be unequal size. First we drop last
+      // rows of matrices and make them the same size.
+      int32 min_length = std::numeric_limits<std::int32_t>::max();;
+      for (size_t i = 0; i < uttlist.size(); i++) {
+        std::string utt = uttlist[i];
+        if (matrix_reader.HasKey(utt)) {
+          min_length = std::min(min_length, matrix_reader.Value(utt).NumRows());
+        }
+      }
+      for (size_t i = 0; i < uttlist.size(); i++) {
+        std::string utt = uttlist[i];
+        if (!matrix_reader.HasKey(utt)) {
+          KALDI_WARN << "No matrix present in input for utterance " << utt;
+          num_utt_err++;
+        } else {
+          if (utt_count == 0)
+            spk_max.Resize(min_length, matrix_reader.Value(utt).NumCols());
+          if (matrix_reader.Value(utt).NumRows() > min_length) {
+            KALDI_WARN << "Cropping rows of matrix " << utt << " from "
+                        << matrix_reader.Value(utt).NumRows() << " to " 
+                        << min_length;
+          
+          }
+          SubMatrix<BaseFloat> temp(matrix_reader.Value(utt), 0, min_length, 
+              0, matrix_reader.Value(utt).NumCols());
+          spk_max.Max(temp);
+          num_utt_done++;
+          utt_count++;
+        }
+      }
+      if (utt_count == 0) {
+        KALDI_WARN << "Not producing output for speaker " << spk
+                    << " since no utterances had matrices";
```

Comment:
nit: extra space
```suggestion
        KALDI_WARN << "Not producing output for speaker " << spk
                   << " since no utterances had matrices";
```